controller.go

```diff
@@ -80,11 +81,17 @@ func (fn HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if err := fn(w, r); err != nil {
 		log.Println("Error: ", err)
 		status := http.StatusInternalServerError
+		var response apiv1.ErrorMsg
```

Comment:
rather than using local vars like status and response to be modified down below, can we have something like?

if errors.As(err, &e) {
  replyJSON(w, e.JSONResponse(), e.StatusCode)
} else {
  replyJSON(w, apiv1.ErrorMsg{Error: "Internal Server Error"}, http.StatusInternalServerError)
}

I think it's easier to follow.

Reply:
done